### PR TITLE
[ios] fix hold and drag spacebar does not move cursor when obscureTex…

### LINKED ISSUE #122139: [iOS] Hold and drag spacebar does not move cursor when `obscureText` is true.
## Steps to Reproduce

1. Create a new application with a `TextFormField` or `TextField` and set `obscureText` to `true` as in the Code Sample below
2. Run the application on an iOS device or simulator
3. Long press on the keybord's spacebar and try to move the cursor

**Expected results:** 
Should be possible to move the cursor using the spacebar

**Actual results:** 
- When `obscureText` is `true` before the TextField is focused, the cursor does not move
- When `obscureText` is `false` before the TextField is focused, the cursor moves as expected

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _controller = TextEditingController();
  bool _obscureText = true;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Flutter Demo')),
      body: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 20),
        child: Column(
          children: [
            const SizedBox(height: 80),
            TextFormField(
              controller: _controller,
              obscureText: _obscureText,
              decoration: InputDecoration(
                suffixIcon: IconButton(
                  icon: Icon(
                    _obscureText ? Icons.visibility_off : Icons.visibility,
                  ),
                  onPressed: () {
                    setState(() {
                      _obscureText = !_obscureText;
                    });
                  },
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.5, on macOS 13.2.1 22D68 darwin-arm64, locale
    en-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.76.0)
[✓] Connected device (5 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

https://user-images.githubusercontent.com/31525418/223569075-d1d1fb96-1fb2-495d-9832-3ef7c21fc274.mp4

### PR DESCRIPTION
**Pull request by LinXunFeng**
Fixes #122139 with engine pr [flutter/engine#40216](https://github.com/flutter/engine/pull/40216)

### CODE REVIEW

**Comment by goderbauer** [comment]
(triage) @justinmc @LongCatIsLooong Could one of you take a look at this one?

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -3072,6 +3072,41 @@ void main() {
...
+          setState = stateSetter;
+          return EditableText(
+            obscureText: obscureText,
+            controller: controller,
+            backgroundCursorColor: Colors.grey,
+            focusNode: focusNode,
+            style: textStyle,
+            cursorColor: cursorColor,
+          );
+        }),
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:3093]: The indentation is strange here, I think this is right:

```suggestion
        home: StatefulBuilder(
          builder: (BuildContext context, StateSetter stateSetter) {
            setState = stateSetter;
            return EditableText(
              obscureText: obscureText,
              controller: controller,
              backgroundCursorColor: Colors.grey,
              focusNode: focusNode,
              style: textStyle,
              cursorColor: cursorColor,
            );
          },
        ),
```

**Review by justinmc — APPROVED** [review]
LGTM with a small nit about indentation.  Thanks for fixing this!

Also just for anyone looking at this PR in the future, the long term plan ([as discussed](https://github.com/flutter/engine/pull/40216#issuecomment-1500440553) on the engine PR) is to sync all of the TextInputConfiguration with the engine.  This is just a stopgap for obscureText specifically.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 1 above)
Comment (LinXunFeng) [reply packages/flutter/test/widgets/editable_text_test.dart:3093]: Thx, you are right.

**Review by cyanglaz — APPROVED** [review]
LGTM. The tests are failing tho, could you rebase and let the ci run again?

**Comment by LinXunFeng** [comment]
> LGTM. The tests are failing tho, could you rebase and let the ci run again?

Done, but the `cla/google` test is still failing.

@justinmc  Hi, your email is not associated with your [CLA](https://cla.developers.google.com/clas), could you please associate it ?

**Comment by justinmc** [comment]
Fixed!  Sorry, it ended up with my personal email.

**Comment by LinXunFeng** [comment]
@cyanglaz Hi, `Google testing` is always pending, could you trigger a re-run? I think this should solve it.

**Comment by Renzo-Olivares** [comment]
cc @Renzo-Olivares from triage, run tests locally.

**Comment by Renzo-Olivares** [comment]
Tests pass locally so I set the testing status to passed manually.

**Pull request merged**